### ISSUE TITLE
Rule to enforce consistent indentation to multiline conditions

### ISSUE THREAD

**Issue by AbdelrahmanHafez** [eslint/eslint#13219]
**Please describe what the rule should do:**
AFAIK, there is currently no way to enforce consistent indentation for multiline-conditions, the rule should do that.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
// passes
if (
  user.isPremium &&
  user.paidAmount > 30 &&
  request.customerType === 'individual' 
) {
  // do stuff
}

if (
  user.isPremium &&
      user.paidAmount > 30 && // warns, too many spaces
request.customerType === 'individual' // warns, too few spaces
) {
  // do stuff
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Because it's not specific to a framework/runtime, it's a general JS styling preference.

**Are you willing to submit a pull request to implement this rule?**
I don't know my way around the ESLint codebase, but I don't mind trying.

**Comment by anikethsaha** [comment]
it can be an option for `indent` instead of a separate rule.

**Comment by mdjermanovic** [comment]
@AbdelrahmanHafez thanks for the issue!

I agree with @anikethsaha, this should be a new option in `indent` rather than a separate rule.

There is already an open proposal #12248 for this enhancement (`MultilineExpression` option).

**Comment by AbdelrahmanHafez** [comment]
Oh, I didn't see that. So let's close this issue and follow progress on #12248.
